pageserver/src/page_service.rs

```diff
@@ -340,6 +358,10 @@ impl TimelineHandles {
                 }
             })
     }
+
+    fn tenant_id(&self) -> Option<TenantId> {
+        self.wrapper.tenant_id.get().cloned()
+    }
```

Comment:
nit: I dont think there would be any problems making this `Option<&TenantId>` or using `copied`

Reply:
https://github.com/neondatabase/neon/pull/9130/commits/4996e4c37d0c52a9a290de23113ae27a7613829f